### PR TITLE
Fix how we query MPI Profile for first name

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR fixes a mistake in how we get user profile info from MPI. I had previously thought it looked like this:
```ruby
MPI::Service.new.find_profile_by_identifier returns mpi_profile
mpi_profile = { first_name: 'Joe' }
```
but it is actually like this
```ruby
MPI::Service.new.find_profile_by_identifier returns mpi_response
mpi_response = { profile: { given_names: ['Joe'] } }
```
This PR adds even more defensive programming in case things go wrong so we can run the whole nightly job, even if some things are missing (logging them instead of throwing an exception).

### CODE REVIEW

**Review by pennja — APPROVED** [review]
Looks good, great work! 👍

**Review by LindseySaari — APPROVED** [review]

File: modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb [hunk 1]
```diff
@@ -140,17 +144,21 @@ def send_email_now(template_id, data)
...
     def get_first_name
       if user_account
-        mpi_profile = MPI::Service.new.find_profile_by_identifier(identifier_type: 'ICN', identifier: user_account.icn)
-        if mpi_profile
-          raise mpi_profile.error if mpi_profile.error
-          raise 'First name not found in MPI profile' unless mpi_profile.first_name
+        mpi_response = MPI::Service.new.find_profile_by_identifier(identifier_type: 'ICN', identifier: user_account.icn)
+        if mpi_response
+          error = mpi_response.error
+          Rails.logger.error('MPI response error', { error: }) if error
```
Comment (bosawt) [inline modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb:150]: Why did you remove the `raise` if there's an MPI error?
